unity/unity-embed-host/CoreCLRHost.cs

```diff
@@ -500,6 +500,32 @@ public static IntPtr unity_method_get_function_pointer([NativeCallbackType("Mono
         return IntPtr.Zero;
     }
 
+    [return: NativeCallbackType("char*")]
+    public static IntPtr method_full_name(
+        [NativeCallbackType("MonoMethod*")] IntPtr method,
+        bool signature)
+    {
+        MethodBase mb = MethodBase.GetMethodFromHandle(method.MethodHandleFromHandleIntPtr());
+        StringBuilder retVal = null;
+        if (mb != null)
+        {
+            retVal = new StringBuilder($"{mb.ReflectedType?.Namespace}.{mb.ReflectedType?.Name}:{mb.Name}");
+            if (signature)
+            {
+                ParameterInfo[] param = mb.GetParameters();
+                retVal.Append(" (");
+                if (param.Length > 0)
+                {
+                    retVal.Append($"{param[0].ParameterType.FullName}");
+                    for (int i = 1; i < param.Length; i++)
+                        retVal.Append($",{param[i].ParameterType.FullName}");
+                }
+                retVal.Append(")");
+            }
+        }
+        return Marshal.StringToHGlobalAnsi(retVal?.ToString()??"");
```

Comment:
This is an allocation that will be freed by the client. We may need more to ensure that all allocations from the embedding API are using the same allocator and deallocator. 

Previously `_strdup` was used in C++ and freed with `mono_unity_g_free` that uses `free`.

So, I think we either need to:
1. Allocate this string with malloc/strdup
2. Migrate all allocating embedding APIs at once to ensure matching allocator/deallocator.